### PR TITLE
Fix the expiration in the SSO sessions stored in the Redis ticket registry

### PR DESCRIPTION
**Pull request by leleuj**
By using a sorted set.

The build passes on my fork: https://github.com/leleuj/cas/pull/3

### CODE REVIEW

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java [hunk 1]
```diff
@@ -288,6 +290,49 @@ void verifyDifferentLoginSamePrincipal() throws Throwable {
...
+    @SpringBootTest(
+            classes = {
+                    CasRedisCoreAutoConfiguration.class,
+                    CasRedisTicketRegistryAutoConfiguration.class,
+                    BaseTicketRegistryTests.SharedTestConfiguration.class
+            }, properties = {
+            "cas.ticket.tgt.core.only-track-most-recent-session=false",
+            "cas.ticket.registry.redis.host=localhost",
+            "cas.ticket.registry.redis.port=6379",
+            "cas.ticket.registry.redis.crypto.enabled=false"
```
Comment (mmoayyed) [inline support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:303]: Formatting here seems a bit off?

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java [hunk 2]
```diff
@@ -288,6 +290,49 @@ void verifyDifferentLoginSamePrincipal() throws Throwable {
...
+        @Qualifier("ticketRedisTemplate")
+        private CasRedisTemplate<String, RedisTicketDocument> ticketRedisTemplate;
+
+        @Test
+        void verifyDifferentLoginSamePrincipal() throws Throwable {
+            val principalId = UUID.randomUUID().toString();
+            for (int i = 0; i < 5; i++) {
+                addTicketAndWait(principalId);
+            }
+            assertEquals(1, ticketRedisTemplate.boundZSetOps("CAS_PRINCIPAL:" + principalId).size());
```
Comment (mmoayyed) [inline support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:320]: Please use `RedisCompositeKey`'s relevant methods for building keys.

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java [hunk 3]
```diff
@@ -288,6 +290,49 @@ void verifyDifferentLoginSamePrincipal() throws Throwable {
...
+        private TicketRegistry ticketRegistry;
+
+        @Autowired
+        @Qualifier("ticketRedisTemplate")
+        private CasRedisTemplate<String, RedisTicketDocument> ticketRedisTemplate;
+
+        @Test
+        void verifyDifferentLoginSamePrincipal() throws Throwable {
+            val principalId = UUID.randomUUID().toString();
+            for (int i = 0; i < 5; i++) {
```
Comment (mmoayyed) [inline support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:317]:
```suggestion
            for (var i = 0; i < 5; i++) {
```

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java [hunk 4]
```diff
@@ -288,6 +290,49 @@ void verifyDifferentLoginSamePrincipal() throws Throwable {
...
+
+        @SneakyThrows
+        private void addTicketAndWait(final String principalId) {
+            val authentication = CoreAuthenticationTestUtils.getAuthentication(principalId);
+            val tgtId = new TicketGrantingTicketIdGenerator(10, StringUtils.EMPTY)
+                    .getNewTicketId(TicketGrantingTicket.PREFIX);
+            val tgt = new TicketGrantingTicketImpl(tgtId, authentication, new HardTimeoutExpirationPolicy(5));
+            ticketRegistry.addTicket(tgt);
+
+            Thread.sleep(4000);
```
Comment (mmoayyed) [inline support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:332]: need to find a better way of controlling time. This test is basically costing us approx $1 every time this test runs.

**Review by mmoayyed — CHANGES_REQUESTED** [review]

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java (hunk 1 above)
Comment (leleuj) [reply support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:303]: Yes. I will fix that.

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java (hunk 3 above)
Comment (leleuj) [reply support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:317]: I will change that.

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java (hunk 2 above)
Comment (leleuj) [reply support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:320]: OK. I will change that.

File: support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java (hunk 4 above)
Comment (leleuj) [reply support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java:332]: Sure. I will revisit the test.

**Comment by leleuj** [comment]
We should be good now.

**Comment by leleuj** [comment]
Backported to v7.0.: https://github.com/apereo/cas/pull/5999

**Pull request merged**